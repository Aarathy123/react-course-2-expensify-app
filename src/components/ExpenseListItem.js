import React from 'react'
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({description,amount,createdAt,id}) => (
    <div>
    <Link to={`/edit/${id}`}>
        <h3>Description : {description}</h3>
        <p>
            {numeral((amount)/100).format("$0.00")} 
            - 
            {moment(createdAt).format("MMMM Do, YYYY")}
        </p>
        </Link>
    </div>
)

export default ExpenseListItem;
