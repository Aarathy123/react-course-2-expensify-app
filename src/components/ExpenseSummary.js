import React from 'react'
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'
import numeral from 'numeral';

export const ExpenseSummary =  ({expenseCount, expenseTotal}) => {
    const expenseWord = expenseCount == 1 ? 'expense' : 'expenses'; 
    return (
        <p>Viewing {expenseCount} {expenseWord} 
        <span>
        {(expenseCount) > 0 && (` totalling ${numeral(expenseTotal/100).format('$0.00')}`)}
        </span>
        </p>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: selectExpensesTotal(visibleExpenses)
    }
};
export default connect(mapStateToProps)(ExpenseSummary);