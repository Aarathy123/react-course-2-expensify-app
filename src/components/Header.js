import {NavLink} from 'react-router-dom';
import React from 'react';

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <div>
            <NavLink exact activeClassName="is-active" to="/"> Home </NavLink>
            <NavLink activeClassName="is-active"  to="/create"> Create </NavLink>
            <NavLink activeClassName="is-active"  to="/edit"> Edit </NavLink>
            <NavLink activeClassName="is-active"  to="/help"> Help </NavLink>
        </div>
    </header>
)

export default Header;