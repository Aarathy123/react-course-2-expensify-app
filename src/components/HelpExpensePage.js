import React from 'react';

const HelpExpensePage = () => (
    <div>
    This is Help Expense Page
    </div>
)

export default HelpExpensePage;