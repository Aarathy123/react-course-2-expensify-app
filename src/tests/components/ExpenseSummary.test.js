import React from 'react';
import { shallow } from 'enzyme';
import {ExpenseSummary} from "../../components/ExpenseSummary";

test('should render correctly expensesSummary with one Expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={100}/>)
    expect(wrapper).toMatchSnapshot();
})

test('should render correctly expensesSummary with multiple Expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={3} expenseTotal={10900}/>)
    expect(wrapper).toMatchSnapshot();
})