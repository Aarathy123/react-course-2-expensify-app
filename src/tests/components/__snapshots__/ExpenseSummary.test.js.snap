// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly expensesSummary with multiple Expenses 1`] = `
<p>
  Viewing 
  3
   
  expenses
  <span>
     totalling $109.00
  </span>
</p>
`;

exports[`should render correctly expensesSummary with one Expense 1`] = `
<p>
  Viewing 
  1
   
  expense
  <span>
     totalling $1.00
  </span>
</p>
`;
